File: bindings/expo-cpu-scanner/ios/ExpoCpuScannerModule.swift
import ExpoModulesCore

public class ExpoCpuScannerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoCpuScanner")

    Function("scanLuma") {
      (
        luma: Data,
        width: Int,
        height: Int,
        stride: Int,
        maxDim: Int,
        refine: Bool
      ) -> String in
      return try QrkBridge.scanLuma(
        luma: luma,
        width: width,
        height: height,
        stride: stride,
        maxDim: maxDim,
        refine: refine
      )
    }

    Function("destroyScanner") {
      // Stateless v1 API — reserved for a future ScanSession.
    }
  }
}

File: bindings/expo-cpu-scanner/src/ExpoCpuScannerModule.ts
import { NativeModule, requireNativeModule } from "expo";

declare class ExpoCpuScannerModule extends NativeModule {
  /**
   * Scan a tightly packed or strided 8-bit luma plane.
   * Returns a JSON string matching `QrScanResult`.
   */
  scanLuma(
    luma: Uint8Array,
    width: number,
    height: number,
    stride: number,
    maxDim: number,
    refine: boolean,
  ): string;

  /** Reserved for a future session API; currently a no-op. */
  destroyScanner(): void;
}

export default requireNativeModule<ExpoCpuScannerModule>("ExpoCpuScanner");
